src/Server/RemoteFSHandler.cpp

```diff
@@ -0,0 +1,543 @@
+        case RemoteFSProtocol::RemoveFile:
+            receivePath(path);
+            disk->removeFile(path);
+            writeVarUInt(RemoteFSProtocol::RemoveFile, *out);
+            out->next();
+            break;
+        case RemoteFSProtocol::RemoveFileIfExists:
+            receivePath(path);
+            disk->removeFileIfExists(path);
+            writeVarUInt(RemoteFSProtocol::RemoveFileIfExists, *out);
+            out->next();
+            break;
+        case RemoteFSProtocol::RemoveDirectory:
+            receivePath(path);
+            disk->removeDirectory(path);
+            writeVarUInt(RemoteFSProtocol::RemoveDirectory, *out);
+            out->next();
+            break;
+        case RemoteFSProtocol::RemoveRecursive:
+            receivePath(path);
+            disk->removeRecursive(path);
+            writeVarUInt(RemoteFSProtocol::RemoveRecursive, *out);
+            out->next();
+            break;
+        case RemoteFSProtocol::SetLastModified:
+            receivePath(path);
+            readVarUInt(time_var, *in);
+            disk->setLastModified(path, Poco::Timestamp::fromEpochTime(time_var));
+            writeVarUInt(RemoteFSProtocol::SetLastModified, *out);
+            out->next();
+            break;
+        case RemoteFSProtocol::GetLastModified:
+            receivePath(path);
+            time_var = disk->getLastModified(path).epochTime();
+            writeVarUInt(RemoteFSProtocol::GetLastModified, *out);
+            writeVarUInt(time_var, *out);
+            out->next();
+            break;
+        case RemoteFSProtocol::GetLastChanged:
+            receivePath(path);
+            time_var = disk->getLastChanged(path);
+            writeVarUInt(RemoteFSProtocol::GetLastChanged, *out);
+            writeVarUInt(time_var, *out);
+            out->next();
+            break;
+        case RemoteFSProtocol::SetReadOnly:
+            receivePath(path);
+            disk->setReadOnly(path);
+            writeVarUInt(RemoteFSProtocol::SetReadOnly, *out);
+            out->next();
+            break;
+        case RemoteFSProtocol::CreateHardLink:
+            receivePath(path);
+            receivePath(path2);
+            disk->createHardLink(path, path2);
+            writeVarUInt(RemoteFSProtocol::CreateHardLink, *out);
+            out->next();
+            break;
+        case RemoteFSProtocol::TruncateFile:
+            receivePath(path);
+            readVarUInt(size_t_var, *in);
+            disk->truncateFile(path, size_t_var);
+            writeVarUInt(RemoteFSProtocol::TruncateFile, *out);
+            out->next();
+            break;
+        default:
+            throw Exception(ErrorCodes::UNKNOWN_PACKET_FROM_CLIENT, "Unknown packet {} from client", toString(packet_type));
+    }
+}
+
+void RemoteFSHandler::receivePath(std::string & path)
+{
+    readStringBinary(path, *in);
+    LOG_TRACE(log, "Received path {}", path);
+}
+
+void RemoteFSHandler::receiveUnexpectedHello()
```

Comment:
this method is a bit strange, why not throw "unexpected packet from client" immediately?